internal/metrics/prometheus/factory.go

```diff
@@ -129,8 +141,21 @@ func (f *Factory) Counter(options metrics.Options) metrics.Counter {
 		Help: help,
 	}
 	cv := f.cache.getOrMakeCounterVec(opts, labelNames)
+	labelValues := f.tagsAsLabelValues(labelNames, tags)
+	metric, err := cv.GetMetricWithLabelValues(labelValues...)
+	if err != nil {
+		if f.logger != nil {
```

Comment:
this is a bad idea. The common practice is to require that a non-nil logger is provided when constructing the object (e.g. by logging something in the constructor like `debug(creating prometheus metrics store)`). There are not that many places in the code where factory is constructed (more so in test than in prod code).